From c379b0ee29dc74e9bd3376d60ca6cd309628e3fb Mon Sep 17 00:00:00 2001
From: Julien Stephan <jstephan@baylibre.com>
Date: Wed, 9 Apr 2025 15:42:12 +0200
Subject: [PATCH 1/2] skip test_nan_inputs[sph_harm]

test_nan_inputs[sph_harm] fails on riscv64 with the following error:

  __________________________________________ test_nan_inputs[sph_harm] __________________________________________
  opt/python/cp310-cp310/lib/python3.10/site-packages/scipy/special/tests/test_nan_inputs.py:57: in test_nan_inputs
      assert_array_equal(np.isnan(res), True, err_msg=msg)
          args       = (nan, nan, nan, nan)
          func       = <ufunc 'sph_harm'>
          msg        = 'got 0j instead of nan'
          res        = np.complex128(0j)
          sup        = <numpy.testing._private.utils.suppress_warnings object at 0x7a1962488d60>
  opt/python/cp310-cp310/lib/python3.10/site-packages/numpy/_utils/__init__.py:85: in wrapper
      return fun(*args, **kwargs)
  E   AssertionError:
  E   Arrays are not equal
  E   got 0j instead of nan
  E   Mismatched elements: 1 / 1 (100%)
  E    ACTUAL: array(False)
  E    DESIRED: array(True)
          args       = (np.False_, True)
          dep_version = '2.0.0'
          fun        = <function assert_array_equal at 0x7a19ad80f0a0>
          kwargs     = {'err_msg': 'got 0j instead of nan'}
          new_name   = 'desired'
          new_names  = ['actual', 'desired']
          old_name   = 'y'
          old_names  = ['x', 'y']

Upstream-Status: Issue [https://github.com/scipy/scipy/issues/22753]

Signed-off-by: Julien Stephan <jstephan@baylibre.com>
---
 tools/wheels/cibw_test_command.sh | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/tools/wheels/cibw_test_command.sh b/tools/wheels/cibw_test_command.sh
index c9dc71ae1..a6a217674 100644
--- a/tools/wheels/cibw_test_command.sh
+++ b/tools/wheels/cibw_test_command.sh
@@ -12,4 +12,8 @@ if [[ $FREE_THREADED_BUILD == "True" ]]; then
 
 fi
 
-python -c "import sys; import scipy; sys.exit(not scipy.test())"
+# test_nan_input fails on riscv64 so skip it
+# see: https://github.com/scipy/scipy/issues/22753
+SKIP_TESTS="not ( test_nan_inputs[sph_harm] )"
+
+python -c "import sys; import scipy; sys.exit(not scipy.test(extra_argv=['-k' , '${SKIP_TESTS}']))"
-- 
2.49.0

